packages/api/src/external/carequality/command/update-patient-discovery-status.ts

```diff
@@ -39,6 +41,18 @@ export async function updatePatientDiscoveryStatus({
     if (status === "completed") {
       docQueryRequestIdToTrigger = getCQData(externalData)?.scheduledDocQueryRequestId;
       newScheduledDocQueryRequestId = undefined;
+    } else if (status === "failed") {
+      newScheduledDocQueryRequestId = undefined;
+      const scheduledDocQueryRequestId = getCQData(externalData)?.scheduledDocQueryRequestId;
+
+      if (scheduledDocQueryRequestId) {
+        setDocQueryProgress({
```

Comment:
- this triggers an update on the patient on a new DB tx inside DB tx;
- this is `async`, we either await it or add asynchronous error handling code.

Ideally we would have a function called inside `setDocQueryProgress` to define the data to be updated on the patient, so we could call it here too.

But if this is too much trouble now, let's at least add a comment bringing attention to this and add the async error handing code to the `setDocQueryProgress` call.